Kickstarter-iOS/Views/Controllers/PledgeViewController.swift

```diff
@@ -26,12 +53,153 @@ final class PledgeViewController: UIViewController {
+  }
+
+  // MARK: - Styles
+
+  override func bindStyles() {
+    super.bindStyles()
+
+    _ = self.view
+      |> checkoutBackgroundStyle
+
+    _ = self.rootScrollView
+      |> rootScrollViewStyle
+
+    _ = self.rootStackView
+      |> rootStackViewStyle
+  }
+
+  // MARK: - View model
+
+  override func bindViewModel() {
+    super.bindViewModel()
+
+    self.viewModel.outputs.configureWithPledgeViewData
+      .observeForUI()
+      .observeValues { [weak self] data in
+        self?.pledgeDescriptionViewController.configureWith(value: data.reward)
+        self?.pledgeAmountViewController.configureWith(value: (data.project, data.reward))
+        self?.pledgeShippingLocationViewController.configureWith(value: (data.project, data.reward))
+        self?.pledgePaymentMethodsViewController.configureWith(value: [GraphUserCreditCard.template])
+      }
+
+    self.viewModel.outputs.configureSummaryViewControllerWithData
+      .observeForUI()
+      .observeValues { [weak self] project, pledgeTotal in
+        self?.pledgeSummaryViewController.configureWith(value: (project, pledgeTotal))
+      }
+
+    self.sessionStartedObserver = NotificationCenter.default
+      .addObserver(forName: .ksr_sessionStarted, object: nil, queue: nil) { [weak self] _ in
+        self?.viewModel.inputs.userSessionStarted()
+      }
+
+    Keyboard.change
+      .observeForUI()
+      .observeValues { [weak self] change in
+        self?.rootScrollView.handleKeyboardVisibilityDidChange(change)
+      }
+
+    self.pledgeShippingLocationViewController.view.rac.hidden
+      = self.viewModel.outputs.shippingLocationViewHidden
+    self.pledgeContinueViewController.view.rac.hidden = self.viewModel.outputs.continueViewHidden
+    self.pledgePaymentMethodsViewController.view.rac.hidden = self.viewModel.outputs.paymentMethodsViewHidden
+  }
+
+  // MARK: - Actions
+
+  @objc func dismissKeyboard() {
+    self.view.endEditing(true)
+  }
+}
+
+extension PledgeViewController: PledgeAmountViewControllerDelegate {
+  func pledgeAmountViewController(
+    _: PledgeAmountViewController,
+    didUpdateAmount amount: Double
+  ) {
+    self.viewModel.inputs.pledgeAmountDidUpdate(to: amount)
+  }
+}
+
+extension PledgeViewController: PledgeShippingLocationViewControllerDelegate {
+  func pledgeShippingLocationViewController(
+    _: PledgeShippingLocationViewController,
+    didSelectShippingRule shippingRule: ShippingRule
+  ) {
+    self.viewModel.inputs.shippingRuleSelected(shippingRule)
+  }
+}
+
+// MARK: - Styles
+
+private let rootScrollViewStyle: ScrollStyle = { scrollView in
+  scrollView
+    |> UIScrollView.lens.showsVerticalScrollIndicator .~ false
+    |> \.alwaysBounceVertical .~ true
+}
+
+private let rootStackViewStyle: StackViewStyle = { stackView in
+  stackView
+    |> \.layoutMargins .~ .init(topBottom: Styles.grid(2), leftRight: Styles.grid(3))
+    |> \.isLayoutMarginsRelativeArrangement .~ true
+    |> \.axis .~ .vertical
+    |> \.distribution .~ .fill
```

Comment:
I've noticed many of styles in this PR are not explicit about the type (helping infer the type and therefore hopefully silencing compiler type check warnings)...would it be too much effort fixing that in this PR? Or should we consider doing this as a sweep before merging `feature-native-checkout`?

Reply:
I changed a few of them but prob best to do a sweep of everything in `feature-native-checkout` to make sure we don't miss anything.